src/services/health-check/presenter-result-builder.php

```diff
@@ -0,0 +1,207 @@
+	 */
+	public function set_label( $label ) {
+		$this->label = $label;
+		return $this;
+	}
+
+	/**
+	 * Sets the name for the test that the plugin uses to identify the test.
+	 *
+	 * @param  string $test_identifier The identifier for the health check.
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_test_identifier( $test_identifier ) {
+		$this->test_identifier = $test_identifier;
+		return $this;
+	}
+
+	/**
+	 * Sets the status of the test result to GOOD (green label).
+	 *
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_status_good() {
+		$this->status = self::STATUS_GOOD;
+		return $this;
+	}
+
+	/**
+	 * Sets the status of the test result to RECOMMENDED (orange label).
+	 *
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_status_recommended() {
+		$this->status = self::STATUS_RECOMMENDED;
+		return $this;
+	}
+
+	/**
+	 * Sets the status of the test result to CRITICAL (red label).
+	 *
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_status_critical() {
+		$this->status = self::STATUS_CRITICAL;
+		return $this;
+	}
+
+	/**
+	 * Sets a description for the test result. This will be the heading for the result in the user interface.
+	 *
+	 * @param  string $description The description for the test result.
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_description( $description ) {
+		$this->description = $description;
+		return $this;
+	}
+
+	/**
+	 * Sets a text that describes how the user can solve the failed health check.
+	 *
+	 * @param  string $actions The descriptive text.
+	 * @return Presenter_Result_Builder This builder.
+	 */
+	public function set_actions( $actions ) {
+		$this->actions = $actions;
+		return $this;
+	}
+
+	/**
+	 * Builds an array of strings in the format that WordPress uses to display health checks (https://developer.wordpress.org/reference/hooks/site_status_test_result/).
+	 *
+	 * @return string[]
+	 */
+	public function build() {
+		return [
+			'label'       => $this->label,
+			'status'      => $this->status,
+			'badge'       => $this->get_badge(),
+			'description' => $this->description,
+			'actions'     => $this->get_actions_with_signature(),
+			'test'        => $this->test_identifier,
+		];
+	}
+
+	/**
+	 * Generates a badge that the user can see.
+	 *
+	 * @return string[]
+	 */
+	private function get_badge() {
+		return [
+			'label' => $this->get_badge_label(),
+			'color' => $this->get_badge_color(),
+		];
+	}
+
+	/**
+	 * Generates the label for a badge.
+	 *
+	 * @return string
+	 */
+	private function get_badge_label() {
+		return __( 'SEO', 'wordpress-seo' );
+	}
+
+	/**
+	 * Generates the color for the badge using the current status.
+	 *
+	 * @return string
+	 */
+	private function get_badge_color() {
+		if ( $this->status === self::STATUS_CRITICAL || $this->status === self::STATUS_RECOMMENDED ) {
```

Comment:
error.
misschien moeten we de status en de bijbehorende presentatie van die status, van elkaar lostrekken; zodat je hier WEL status === xyz kunt doen.

Reply:
not true - het zijn interne WP consts